contracts/counter/README.md

```diff
@@ -0,0 +1,255 @@
+        ...
+    }
+```
+
+Be defining this attribute, you can now use : 
+```rust,ignore
+    money_market.deposit_stable(coins(456, "ujunox"))?;
+```
+
+#### fn_name
+Be careful what you name your enum variants, you could stumble upon one of the reserved keywords of cw-orch. If this happens, you can use the fn_name attribute to rename a generated function.
+
+```rust,ignore
+
+    ExecuteMsg{
+        #[fn_name("proxy_execute")]
+        Execute{
+            msg: CosmoMsg
+        }
+    }
+
+    // Will error because the execute function is reserved for contract execution. Will not even compile actually
+    money_market.execute(message_to_execute_via_a_proxy)?;
+```
+
+```rust,ignore
+    ExecuteMsg{
+        #[fn_name("proxy_execute")]
+        Execute{
+            msg: CosmoMsg
+        }
+    }
+    // This works smoothly !
+    money_market.proxy_execute(message_to_execute_via_a_proxy)?;
+```
+
+This is also true for query functions.
+
+
+#### impl_into
+For more details about this attribute, [see the full documentation](https://orchestrator.abstract.money/single_contract/interfaces.html#impl_into-attribute)
+
+
+
+
+## Other cw-orch traits
+
+### Deploy
+
+cw-orchestrator defines the [Deploy](../../cw-orch/src/deploy.rs) trait that allows developers to create an interface that allows themselves or other developers to : 
+1. Simply deploy their codebase on a test chain or elsewhere. [See it live here](https://github.com/AbstractSDK/abstract/blob/main/framework/scripts/src/bin/full_deploy.rs#L29).
+2. Provide bindings (addresses, code_ids), for other developers to integrate with the platform. [See it live here](https://github.com/AbstractSDK/abstract/blob/main/modules/contracts/apps/croncat/examples/deploy.rs#L32).
+
+This is even more powerful than simply exposing the single contract interfaces, because it allows full customization of the deployment procedure. It also allows shipping other custom methods for simplifying the life of the integrator.
+
+Imagine you are a developer that wants to integrate with [DA0-DA0](https://github.com/DA0-DA0). With this deploy trait, you could test your application like so : 
+
+```rust,ignore
+use daodao::core::DaoDao;
+use cw_orch::daemon::networks::JUNO_1;
+
+pub fn test(){
+    let chain = Daemon::builder()
+        .chain(JUNO_1) 
+        .handle(Runtime::new().unwrap().handle()) 
+        .build()
+        .unwrap();
+        
+    let dao_id = 56u64;
+    let daodao = DaoDao::load_from(chain.clone())?;
+    daodao.set_account_id(dao_id)?;
+
+    let my_custom_gov_admin = GovAdmin::new("gov-admin", chain)?;
+    my_custom_gov_admin.upload()?;
+    my_custom_gov_admin.instantiate(&InstantiateMsg{account_id: dao_id});
+
+    daodao.nominate_admin(my_custom_gov_admin.address().to_string())?;
+}
+```
+In our example, all the daodao core addresses are shipped with the daodao crate directly. Projects need to verify certain conditions in order to ship this state with the deploy bundle.
```

Comment:
This first line is not entirely clear. I would add a little elaboration on the definition of "daodao core addresses are shipped with the daodao crate directly."

What does it mean to "ship daodao core addresses"?  the addresses of the contracts deployed according to the latest crate or the addresses of what?